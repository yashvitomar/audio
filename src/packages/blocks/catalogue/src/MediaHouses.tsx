import React from "react";
import {
  Text,
  View,
  Image,
  FlatList,
  Dimensions,
  StyleSheet,
  TouchableOpacity
} from "react-native";
import MediaHousesController, { Props } from "./MediaHousesController";
import Header from "../../../components/src/AppHeader";
import scale, { verticalScale } from '../../../components/src/Scale';
const screenWidth = Dimensions.get("window").width;
const screenHeight = Dimensions.get("window").height;
import { search, theHindu, bellIcon, logo, backArrow, bbcHindi } from "./assets";

import { ScrollView } from "react-native-gesture-handler";
import Scale from "../../../components/src/Scale";
// Customizable Area End

export default class MediaHouses extends MediaHousesController {
  constructor(props: Props) {
    super(props);
    // Customizable Area Start
    // Customizable Area End
  }

  renderPopularMediaHouseCell = (item: any, index: any) => {
    return (
      <View style={{ justifyContent: 'space-between', flex: 1, marginTop: Scale(10) }}>
        <View style={styles.logoView}>
          <TouchableOpacity style={{ alignItems: 'center', justifyContent: "space-between" }} onPress={() => this.props.navigation.navigate('LandingPopularMedia', { title: 'popularMedia', item: item })}>
            <View style={styles.imageViewStyle}>
              <Image source={{ uri: item?.attributes?.image }} style={styles.mediahouselogo} resizeMode={"cover"} />
            </View>
            <Text style={styles.mediahousename}>{item?.attributes?.media_house}</Text>
          </TouchableOpacity>
        </View>
      </View>
    )
  }

  renderPopularMediaHouseContainer = () => {
    return (
      <View style={styles.mediahousecell}>
        <Text style={styles.heading}>Popular Media Houses</Text>
        <FlatList
          // style={{backgroundColor:'red'}}
          data={this.state.mediahouse}
          extraData={this.state}
          numColumns={3}
          renderItem={({ item, index }) =>
            this.renderPopularMediaHouseCell(item, index)
          }
        />
      </View>
    )
  }

  render() {
    return (
      <View style={styles.container}>
        <Header
          backArrow={backArrow}
          logo={logo}
          search={search}
          bell={bellIcon}
          onPress={() => this.props.navigation.goBack()}
          onPressNotification={() => this.props.navigation.navigate('')}
          menu={undefined}
          onPressSearch={() => { }}
          searchScreen={false}
          back={undefined}
          title={undefined}
          onPressBack={() => { }} />
        {this.renderPopularMediaHouseContainer()}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
  },
  imageViewStyle: {
    height: scale(100),
    width: scale(100),
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.34,
    shadowRadius: 3,
    elevation: 4,
    borderRadius: Scale(10)
  },
  mediahouselogo: {
    height: scale(100),
    width: scale(100),
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.34,
    shadowRadius: 3,
    elevation: 4,
    borderRadius: Scale(10)
  },
  logoView: {
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.34,
    shadowRadius: 3,
    borderRadius: Scale(10)
  },
  mediahousecell: {
    marginTop: scale(25),
    flex:1
  },
  mediahousename: {
    fontSize: scale(16),
    marginTop: Scale(5),
    textAlign: 'center'
  },
  heading: {
    fontWeight: 'bold',
    fontSize: scale(18),
    marginHorizontal: Scale(20)
  }
})
