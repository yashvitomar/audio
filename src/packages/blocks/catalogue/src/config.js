Object.defineProperty(exports, "__esModule", {
  value: true
});

// Customizable Area Start
exports.productApiContentType = "application/json";
exports.apiMethodTypeGet = "GET";
exports.productAPiEndPoint = "catalogue/catalogues";
// Customizable Area End