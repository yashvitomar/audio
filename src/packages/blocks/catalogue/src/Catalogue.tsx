import React from "react";
// Customizable Area Start
import {
  StyleSheet,
  Text,
  View,
  Image,
  TouchableOpacity,
  ScrollView,
  FlatList,
  ImageBackground
} from "react-native";
import Icon from 'react-native-vector-icons/SimpleLineIcons';
import { search, navIcon, bellIcon, logo, save } from "./assets";
import Scale from "../../../components/src/Scale";
import Header from "../../../components/src/AppHeader";
import Loader from "../../../components/src/Loader";
import FastImage from 'react-native-fast-image'
// Customizable Area End

import CatalogueController, { Props } from "./CatalogueController";

export default class Catalogue extends CatalogueController {
  constructor(props: Props) {
    super(props);
    // Customizable Area Start
    // Customizable Area End
  }

  // Customizable Area Start
  renderFooter = () => {
    return (
      //Footer View with Load More button
      <View style={styles.footer}>
        {this.state.loading ? (
          <Loader loading={this.state.loading} style={{}} />
          // <ActivityIndicator
          //   // color="white"
          //   style={{ marginLeft: 8 }} />
        ) : null}
      </View>
    );
  }

  renderTrendingMediaHouseItem = ({ item }: any) => (
    <View style={styles.trendingView}>
      <View style={[styles.mediaImageStyleView, { bottom: 0, backgroundColor: '#fff', borderRadius: Scale(10) }]}>
        <FastImage source={{ uri: item.attributes.image }} style={styles.trendingImageStyle} resizeMode='stretch' />
      </View>
    </View>
  );

  renderPopularMediaHouseItem = ({ item }: any) => (
    <View style={styles.popularMediaHouseView}>
      <View style={[styles.mediaImageStyleView, { borderRadius: Scale(10), backgroundColor: '#fff' }]}>
        <Image source={{ uri: item?.attributes?.image }} style={styles.medisImageStyle} resizeMode='stretch' />
      </View>
      <Text style={{ marginTop: Scale(8), alignSelf: 'center', fontWeight: '500', textAlign: 'center', width: '75%' }}>{item?.attributes?.media_house}</Text>
    </View>
  );

  renderCategoryItem = ({ item }: any) => (
    <View style={styles.trendingView}>
      <Image source={{ uri: item?.attributes?.image }} style={[styles.imageStyle, { justifyContent: 'flex-end', alignItems: 'flex-start', height: Scale(110), width: Scale(140), borderRadius: 10 }]} resizeMode='cover' />
      <View style={{ marginTop: Scale(5) }}>
        <Text style={{ color: 'blue', fontWeight: '500', fontSize: Scale(16), marginLeft: Scale(10), bottom: Scale(5) }}>{item.attributes.name}</Text>
        <Text style={{ color: '#000', fontWeight: '500', fontSize: Scale(12), marginLeft: Scale(10), bottom: Scale(5) }}>{item?.attributes?.number_of_articles + ' Articles'}</Text>
      </View>
    </View>
  );

  renderAllArticlesItem = ({ item }: any) => (
    <View style={styles.mediaHouseView}>
      <View style={styles.mediaImageStyleView}>
        <ImageBackground source={{ uri: item?.attributes?.image }} style={styles.imageStyle} imageStyle={{ borderRadius: Scale(10) }} resizeMode='stretch'>
          <TouchableOpacity style={styles.playControlView}>
            <Icon name='control-play' size={10} color='blue' />
          </TouchableOpacity>
        </ImageBackground>
      </View>
      <View style={{ marginHorizontal: Scale(10), width: "76%", backgroundColor: '#fff' }}>
        <View style={{ flexDirection: 'row', alignItems: 'center', backgroundColor: '#fff', justifyContent: 'space-between', }}>
          <Text style={[styles.name, { color: 'blue', opacity: 0.7, width: "94%", backgroundColor: '#fff' }]} numberOfLines={1}>{item?.attributes?.title}</Text>
          <TouchableOpacity>
            <Image source={save} style={{ height: Scale(15), width: Scale(15), tintColor: 'blue' }} resizeMode="contain" />
          </TouchableOpacity>
        </View>
        <Text style={[styles.name, { opacity: 0.7, marginTop: Scale(3), backgroundColor: '#fff' }]} numberOfLines={2}>{item?.attributes?.description}</Text>
        <Text style={styles.mediaName}>{item?.attributes?.media_house}</Text>
        <View style={styles.timeView}>
          <Text style={{ color: '#000' }}>{'4h ago'}</Text>
          <View style={styles.dotView}></View>
          <Text>{'Aug 08, 22'}</Text>
          <View style={styles.dotView}></View>
          <Text>{item?.attributes?.country}</Text>
        </View>
      </View>
    </View>
  );

  renderTrendingMediaHouse = () => {
    const { trendingMediaHouse } = this.state;
    return (
      <View style={styles.trendingMediaView}>
        <FlatList
          style={{}}
          data={trendingMediaHouse.length > 3 ? trendingMediaHouse.slice(0, 3) : trendingMediaHouse}
          horizontal={true}
          showsHorizontalScrollIndicator={false}
          renderItem={(item) => this.renderTrendingMediaHouseItem(item)}
        />
      </View>
    )
  }

  renderPopularMediaHouse = () => {
    const { popularMediaHouseArray } = this.state;
    return (<View style={styles.popularMediaHouseView}>
      <FlatList
        style={{}}
        data={popularMediaHouseArray.length > 3 ? popularMediaHouseArray.slice(0, 3) : popularMediaHouseArray}
        horizontal={true}
        showsHorizontalScrollIndicator={false}
        renderItem={(item) =>  this.renderPopularMediaHouseItem(item)}
      />
    </View>)
  }

  renderCategories = () => {
    const { categoryArray } = this.state;
    return (
      <View style={styles.popularMediaHouseView}>
        <FlatList
          style={{}}
          data={categoryArray.length > 3 ? categoryArray.slice(0, 3) : categoryArray}
          horizontal={true}
          showsHorizontalScrollIndicator={false}
          renderItem={(item) => this.renderCategoryItem(item)}
        />
      </View>
    )
  }

  renderAllArticles = () => {
    const { allArticleArray } = this.state;
    return (
      <View style={{ flex: 1, marginTop: Scale(10), backgroundColor: '#fff', }}>
        <FlatList
          style={{ marginTop: Scale(5), marginBottom: Scale(80), height: Scale(400) }}
          data={allArticleArray}
          horizontal={false}
          nestedScrollEnabled={true}
          showsHorizontalScrollIndicator={false}
          renderItem={(item) => this.renderAllArticlesItem(item)}
          onMomentumScrollEnd={() => {
            if (!this.state.onEndReachedCalledDuringMomentum) {
              this.onEndReached();    // LOAD MORE DATA
              this.setState({ onEndReachedCalledDuringMomentum: true })
            }
          }}
          onEndReached={() => this.setState({ onEndReachedCalledDuringMomentum: false })}
          ListFooterComponent={() => this.renderFooter()}
        />
      </View>
    )
  }
  // Customizable Area End


  render() {
    const { navigation } = this.props;
    const { trendingMediaHouse, popularMediaHouseArray, categoryArray, allArticleArray } = this.state;

    return (
      <>
        {/* Merge Engine DefaultContainer */}
        {/* Customizable Area Start */}
        {/* Merge Engine UI Engine Code */}
        <View style={styles.mainContainer}>
          <Header
            menu={navIcon}
            logo={logo}
            search={search}
            bell={bellIcon}
            onPress={() => this.props.navigation.openDrawer()}
            onPressNotification={() => this.props.navigation.navigate('Pushnotifications')}
            onPressSearch={() => { }}
            searchScreen={false}
            back={undefined}
            title={undefined}
            backArrow={undefined}
            onPressBack={() => { }} />
          <ScrollView nestedScrollEnabled={true}>
            <View style={{ flex: 1, }}>
              <View style={styles.seeMoreView}>
                <Text style={styles.popularText}>Trending</Text>
                <TouchableOpacity onPress={() => navigation.navigate('Trending')}>
                  <Text style={styles.seeMoreText}>See more</Text>
                </TouchableOpacity>
              </View>
              {trendingMediaHouse && trendingMediaHouse.length > 0 ? (
                this.renderTrendingMediaHouse()
              ) : (
                <View style={[styles.noDataView, { marginTop: Scale(30) }]}>
                  <Text style={styles.noDataTxt}>No data found!</Text>
                </View>
              )}
              <View style={styles.seeMoreView}>
                <Text style={styles.popularText}>Popular Media Houses</Text>
                <TouchableOpacity onPress={() => navigation.navigate('PopularMediaHouses')}>
                  <Text style={styles.seeMoreText}>See more</Text>
                </TouchableOpacity>
              </View>
              {popularMediaHouseArray && popularMediaHouseArray.length > 0 ? (
                this.renderPopularMediaHouse()
              ) : (
                <View style={[styles.noDataView, { marginTop: Scale(30) }]}>
                  <Text style={styles.noDataTxt}>No data found!</Text>
                </View>
              )}
              <View style={styles.seeMoreView}>
                <Text style={styles.popularText}>Categories</Text>
                <TouchableOpacity onPress={() => navigation.navigate('Categories')}>
                  <Text style={styles.seeMoreText}>See more</Text>
                </TouchableOpacity>
              </View>
              {categoryArray && categoryArray.length > 0 ? (
                this.renderCategories()
              ) : (
                <View style={[styles.noDataView, { marginTop: Scale(30) }]}>
                  <Text style={styles.noDataTxt}>No data found!</Text>
                </View>
              )}
              <View style={styles.seeMoreView}>
                <Text style={styles.popularText}>All Articles</Text>
                <TouchableOpacity onPress={() => { }}>
                  <Text style={styles.seeMoreText}>See more</Text>
                </TouchableOpacity>
              </View>
              {allArticleArray && allArticleArray.length > 0 ? (
                this.renderAllArticles()
              ) : (
                <View style={[styles.noDataView, { marginTop: Scale(30) }]}>
                  <Text style={styles.noDataTxt}>No data found!</Text>
                </View>)}
            </View>
            <Loader loading={this.state.isLoading} />
          </ScrollView>
        </View>
        {/* Merge Engine UI Engine Code */}
        {/* Customizable Area End */}
        {/* //Merge Engine End DefaultContainer */}
      </>
    );
  }
}

// Customizable Area Start
const styles = StyleSheet.create({
  mainContainer: {
    flex: 1,
    backgroundColor: "#fff"
  },
  outerView: {
    flexDirection: 'row',
    marginTop: Scale(15),
    marginHorizontal: Scale(15),
    alignItems: 'center',
    justifyContent: 'space-between'
  },
  smhImage: {
    height: Scale(22),
    width: Scale(22),
    borderRadius: Scale(15),
    left: Scale(110)
  },
  sydneyTxt: {
    fontSize: Scale(16),
    fontWeight: 'bold',
    marginLeft: Scale(5),
    // opacity:0.7
  },
  national: {
    height: Scale(130),
    width: Scale(130),
    borderRadius: Scale(15)
  },
  sliderView: {
    marginHorizontal: Scale(10),
    marginTop: Scale(10)
  },
  slider: {
    width: '100%',
  },
  durationView: {
    flexDirection: 'row',
    marginHorizontal: Scale(25),
    justifyContent: 'space-between',
    alignItems: 'center'
  },
  durationTxt: {
    fontWeight: 'bold',
    color: '#000',
    opacity: 0.5
  },
  musicControlView: {
    flexDirection: 'row',
    marginHorizontal: Scale(30),
    justifyContent: 'space-between',
    alignItems: 'center',
    marginTop: Scale(5)
  },
  speedPrevIcon: {
    height: Scale(22),
    width: Scale(23),
  },
  musicActivitiesView: {
    backgroundColor: '#CBC7C7',
    flexDirection: 'row',
    marginHorizontal: Scale(30),
    justifyContent: 'space-between',
    alignItems: 'center',
    marginTop: Scale(10),
    borderRadius: Scale(20),
    paddingHorizontal: 20,
    padding: 10
  },
  lineView: {
    borderWidth: 0.7,
    opacity: .1,
    elevation: 2,
    borderColor: '#000',
    marginTop: Scale(15)
  },
  sharedView: {
    flexDirection: 'row',
    marginHorizontal: Scale(10),
    marginTop: Scale(5)
  },
  sharedBtn: {
    width: "40%",
    padding: Scale(10),
    backgroundColor: 'blue',
    borderRadius: Scale(10),
    elevation: 2,
    margin: 5,
    justifyContent: 'center',
    alignItems: 'center'
  },
  mediaHouseView: {
    backgroundColor: '#fff',
    marginHorizontal: Scale(15),
    flexDirection: 'row',
    elevation: Scale(5),
    borderRadius: Scale(10),
    padding: Scale(15),
    marginVertical: Scale(5),
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.34,
    shadowRadius: 3,
  },
  imageStyle: {
    height: Scale(80),
    width: Scale(80),
    justifyContent: 'center',
    alignItems: 'center'
  },
  name: {
    fontSize: Scale(16),
    fontWeight: 'bold'
  },
  mediaName: {
    fontSize: Scale(14),
    fontWeight: 'bold',
    opacity: 0.4,
    marginTop: Scale(5)
  },
  timeView: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: "center",
    // width: '85%',
    marginTop: Scale(5),
    // backgroundColor:'red'
  },
  dotView: {
    height: Scale(5),
    width: Scale(5),
    borderRadius: Scale(5),
    backgroundColor: '#000'
  },
  subHeadText: {
    fontSize: Scale(16),
    fontWeight: 'bold',
    marginHorizontal: Scale(15),
    opacity: 0.7
  },
  playControlView: {
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#fff',
    elevation: 3,
    borderRadius: Scale(20),
    padding: Scale(10)
  },
  seeMoreView: {
    flexDirection: 'row',
    marginHorizontal: Scale(20),
    justifyContent: 'space-between',
    alignItems: 'center',
    marginTop: Scale(15)
  },
  popularText: {
    fontSize: Scale(16),
    fontWeight: 'bold',
  },
  seeMoreText: {
    fontSize: Scale(14),
    fontWeight: 'bold',
    color: 'blue'
  },
  songListView: {
    marginHorizontal: Scale(15),
    justifyContent: 'center',
    alignItems: "center",
    marginTop: Scale(10),
  },
  trendingMediaView: {
    marginTop: Scale(10),
  },
  trendingView: {
    // backgroundColor:'red',
    marginHorizontal: Scale(10),
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.34,
    shadowRadius: 2,
    elevation: 5,
  },
  trendingImageStyle: {
    height: Scale(120),
    width: Scale(120),
    borderRadius: Scale(10),
  },
  popularMediaHouseView: {
    alignItems: "center",
    marginHorizontal: Scale(10),
  },
  mediaImageStyleView: {
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.34,
    shadowRadius: 2,
    elevation: 5,
    // backgroundColor:'#fff',
    // borderRadius: Scale(10),
  },
  medisImageStyle: {
    height: Scale(90),
    width: Scale(90),
    borderRadius: Scale(5),
  },
  noDataView: {
    alignSelf: 'center',
    justifyContent: 'center',
    alignItems: 'center',
    // flex: 1
  },
  noDataTxt: {
    color: 'gray',
    fontWeight: 'bold'
  },
  footer: {
    padding: 10,
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
    // backgroundColor:'red',
    bottom: 30
  },
});
// Customizable Area End
