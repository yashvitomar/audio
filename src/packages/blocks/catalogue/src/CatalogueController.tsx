import { IBlock } from "../../../framework/src/IBlock";
import { Message } from "../../../framework/src/Message";
import { BlockComponent } from "../../../framework/src/BlockComponent";
import MessageEnum, {
  getName
} from "../../../framework/src/Messages/MessageEnum";
import { runEngine } from "../../../framework/src/RunEngine";

// Customizable Area Start
import { createRef } from 'react';
import AsyncStorage from "@react-native-async-storage/async-storage";
// Customizable Area End

export const configJSON = require("./config");

export interface Props {
  navigation: any;
  id: string;
  // Customizable Area Start
  // Customizable Area End
}

interface S {
  // Customizable Area Start
  trendingMediaHouse: any;
  popularMediaHouseArray: any;
  allArticleArray: any;
  categoryArray: any;
  isMedia: boolean;
  enableScrollViewScroll: boolean
  isLoading: boolean;
  loading: boolean;
  perPage: any;
  page: any;
  onEndReachedCalledDuringMomentum: boolean,
  scrollDisabled: boolean,
  // Customizable Area End
}

interface SS {
  id: any;
}
export default class CatalogueController extends BlockComponent<Props, S, SS> {
  currentRef: any;
  getTrendingMediaApiCallId: string = "";
  getPopularMediaHouseDataApicallId: string = "";
  getCategoriesApiCallId: string = "";
  getAllArticlesApiCallId: string = ""

  constructor(props: Props) {
    super(props);
    this.receive = this.receive.bind(this);

    this.subScribedMessages = [
      getName(MessageEnum.RestAPIResponceMessage),
      // Customizable Area Start
      // Customizable Area End
    ];

    this.currentRef = createRef();

    this.state = {
      trendingMediaHouse: [],
      popularMediaHouseArray: [],
      allArticleArray: [],
      categoryArray: [],
      isMedia: false,
      enableScrollViewScroll: true,
      isLoading: false,
      loading: false,
      perPage: 6,
      page: 1,
      onEndReachedCalledDuringMomentum: true,
      scrollDisabled: false,
    };
    runEngine.attachBuildingBlock(this as IBlock, this.subScribedMessages);
    // Customizable Area Start
    // Customizable Area End
  }

  async componentDidMount() {
    let token = await AsyncStorage.getItem('token');
    let id = await AsyncStorage.getItem('userId');
    this.getTrendingMediaApi(token);
    this.getPopularMediaApi(token, id);
    this.getCategoriesMediaApi(token, id);
    this.getAllArticlesApi();
  }

  async receive(from: string, message: Message) {
    const apiRequestCallId = message.getData(getName(MessageEnum.RestAPIResponceDataMessage));
    const responseJson = message.getData(getName(MessageEnum.RestAPIResponceSuccessMessage));
    const errorReponse = message.getData(getName(MessageEnum.RestAPIResponceErrorMessage));

    switch (apiRequestCallId) {
      case this.getTrendingMediaApiCallId:
        this.handleTrendingMediaApiResponse(responseJson, errorReponse);
        break;
      case this.getPopularMediaHouseDataApicallId:
        this.handlePopularMediaHouseDataApiResponse(responseJson, errorReponse);
        break;
      case this.getCategoriesApiCallId:
        this.handleCategoriesApiResponse(responseJson, errorReponse);
        break;
      case this.getAllArticlesApiCallId:
        this.handleAllArticlesApiResponse(responseJson, errorReponse);
        break;
    }

    if (errorReponse) {
      this.setState({ isLoading: false });
    }
  }

  // Customizable Area Start
  handleTrendingMediaApiResponse(responseJson: any, errorReponse: any) {
    if (responseJson && !responseJson.errors) {
      console.log("TRENDING-MEDIA-RESPONSE", responseJson);
      this.setState({ trendingMediaHouse: responseJson.data })
    } else {
      this.setState({ isLoading: false })
    }
  }

  handlePopularMediaHouseDataApiResponse(responseJson: any, errorReponse: any) {
    if (responseJson && !responseJson.errors) {
      console.log("POPULAR-MEDIA-RESPONSE", responseJson);
      this.setState({
        isLoading: false,
        popularMediaHouseArray: responseJson.media_hosues.data,
      });
    } else {
      this.setState({ isLoading: false })
    }
  }

  handleCategoriesApiResponse(responseJson: any, errorReponse: any) {
    if (responseJson && !responseJson.errors) {
      console.log("CATEGORIES-MEDIA-RESPONSE", responseJson);
      this.setState({
        categoryArray: responseJson.categories.data,
      });
    } else {
      this.setState({ isLoading: false })
    }
  }

  handleAllArticlesApiResponse(responseJson: any, errorReponse: any) {
    if (responseJson && !responseJson.errors) {
      console.log("ALL_ARTICLES-RESPONSE", responseJson);
      let temp: any;
      if (this.state.allArticleArray.length > 0) {
        temp = this.state.allArticleArray.concat(responseJson.articles.data)
      } else {
        temp = responseJson.articles.data
      }
      this.setState({ isLoading: false, loading: false, allArticleArray: temp })
    } else {
      console.log("ALL_ARTICLES-ERROR-RESPONSE");
      this.setState({ isLoading: false, loading: false })
    }
  }

  apiCall = async (data: any) => {
    const { contentType, method, endPoint, body, token } = data;
    console.log("endpoint", endPoint);
    const header = {
      "Content-Type": contentType,
      token: token
      // token: this.state.token,
    };
    const requestMessage = new Message(
      getName(MessageEnum.RestAPIRequestMessage)
    );
    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestHeaderMessage),
      JSON.stringify(header)
    );
    requestMessage.addData(
      getName(MessageEnum.RestAPIResponceEndPointMessage),
      endPoint
    );
    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestMethodMessage),
      method
    );
    {
      body &&
        requestMessage.addData(
          getName(MessageEnum.RestAPIRequestBodyMessage),
          JSON.stringify(body)
        );
    }
    runEngine.sendMessage(requestMessage.id, requestMessage);
    return requestMessage.messageId;
  };

  async getTrendingMediaApi(token: any) {
    this.getTrendingMediaApiCallId = await this.apiCall({
      contentType: "application/json",
      method: "GET",
      endPoint: "bx_block_articles/trending?preference=false",
      token: token
      // body: apiData,
    });
  }

  async getPopularMediaApi(token: any, id: any) {
    this.getPopularMediaHouseDataApicallId = await this.apiCall({
      contentType: "application/json",
      method: "GET",
      endPoint: "bx_block_admin/media_houses",
      // endPoint: "bx_block_admin/media_house_article?id=" + id,
      token: token
      // body: apiData,
    });
  }

  async getCategoriesMediaApi(token: any, id: any) {
    this.setState({ isLoading: true });
    this.getCategoriesApiCallId = await this.apiCall({
      contentType: "application/json",
      method: "GET",
      endPoint: "bx_block_categories/categories",
      // endPoint: "bx_block_categories/category_article?id=" + id,
      token: token
      // body: apiData,
    });
  }
  
  async getAllArticlesApi() {
    let token = await AsyncStorage.getItem('token')
    this.getAllArticlesApiCallId = await this.apiCall({
      contentType: "application/json",
      method: "GET",
      endPoint: "bx_block_articles/articles?per_page=" + this.state.perPage + "&page=" + this.state.page,
      token: token
      // body: apiData,
    });
  }

  onEndReached = async () => {
    console.log("onEndReached")
    this.setState({ page: this.state.page + 1, perPage: this.state.perPage })
    this.setState({ loading: true });
    this.getAllArticlesApi();
  }
  // Customizable Area End
}
