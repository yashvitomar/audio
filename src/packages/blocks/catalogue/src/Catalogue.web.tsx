import React from "react";
// Customizable Area Start
import {
  Box,
  Container,
  Typography,
} from "@material-ui/core";
import { createTheme, ThemeProvider } from "@material-ui/core/styles";
// Customizable Area End

import CatalogueController, { Props } from "./CatalogueController";

const theme = createTheme({
  palette: {
    primary: {
      main: "#0000ff",
      contrastText: "#fff",
    },
  },
});

export default class Catalogue extends CatalogueController {
  constructor(props: Props) {
    super(props);
    // Customizable Area Start
    // Customizable Area End
  }

  // Customizable Area Start
  getWebList(item: any) {
    let value = item.attributes;
    return (
      <Box sx={webStyle.productBox} key={`product-item-${item.id}`}>
        <Box sx={webStyle.ImgContainer}>
          <img style={webStyle.productImg} src={value.images?.[0].url} />
        </Box>
        <Box sx={webStyle.detailContent}>
          <Typography variant="h6">{value.name}</Typography>
          <Typography variant="h6">{value.price}</Typography>
          <Typography variant="h6">{value.average_rating}</Typography>
        </Box>
      </Box>
    );
  }
  // Customizable Area End

  render() {
    return (
      //Merge Engine DefaultContainer
      <ThemeProvider theme={theme}>
        <Container maxWidth="md">
          <Box sx={webStyle.productContainer}>
            {this.state.arrayHolder &&
              this.state.arrayHolder.map((item: any, index: any) =>
                this.getWebList(item)
              )}
          </Box>
        </Container>
      </ThemeProvider>
      //Merge Engine End DefaultContainer
    );
  }
}

// Customizable Area Start
const webStyle = {
  productContainer: {
    display: "flex",
    flexDirection: "row",
    flexWrap: "wrap",
    width: "100%",
    backgroundColor: "white",
    marginTop: "20px",
  },
  productBox: {
    height: 250,
    width: "49%",
    marginRight: "5px",
    flexDirection: "column",
  },
  ImgContainer: {
    //marginBottom: 15,
    height: 150,
  },
  productImg: {
    width: "100%",
    height: "100%",
  },
  detailContent: {
    display: "flex",
    flexDirection: "column",
  },
};
// Customizable Area End
