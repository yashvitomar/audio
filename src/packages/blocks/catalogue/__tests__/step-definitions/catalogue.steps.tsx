import { defineFeature, loadFeature } from "jest-cucumber";
import { shallow, ShallowWrapper } from "enzyme";

import * as helpers from "../../../../framework/src/Helpers";

import React from "react";
import Catalogue from "../../src/Catalogue";
const navigation = require("react-navigation");

const screenProps = {
  navigation: navigation,
  id: "Catalogue",
};

const feature = loadFeature("./__tests__/features/catalogue-scenario.feature");

defineFeature(feature, (test) => {
  beforeEach(() => {
    jest.resetModules();
    jest.doMock("react-native", () => ({ Platform: { OS: "web" } }));
    jest.spyOn(helpers, "getOS").mockImplementation(() => "web");
  });

  test("User navigates to catalogue", ({ given, when, then }) => {
    let exampleBlockA: ShallowWrapper;
    let instance: Catalogue;

    given("I am a User loading catalogue", () => {
      exampleBlockA = shallow(<Catalogue {...screenProps} />);
    });

    when("I navigate to the catalogue", () => {
      instance = exampleBlockA.instance() as Catalogue;
    });

    then("catalogue will load with out errors", () => {
      expect(exampleBlockA).toBeTruthy();
      expect(exampleBlockA).toMatchSnapshot();
    });

    then("I can leave the screen with out errors", () => {
      instance.componentWillUnmount();
      expect(exampleBlockA).toBeTruthy();
      expect(exampleBlockA).toMatchSnapshot();
    });
  });
});
