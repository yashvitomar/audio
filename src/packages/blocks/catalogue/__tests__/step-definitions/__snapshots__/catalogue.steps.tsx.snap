// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`catalogue User navigates to catalogue 1`] = `
<ScrollViewMock
  keyboardShouldPersistTaps="always"
  style={
    Object {
      "backgroundColor": "#ffffffff",
      "flex": 1,
      "marginLeft": "auto",
      "marginRight": "auto",
      "maxWidth": 650,
      "width": "100%",
    }
  }
>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "flex": 1,
              "justifyContent": "center",
              "width": "100%",
            },
            Object {
              "display": "flex",
              "flexDirection": "row",
            },
          ]
        }
      >
        <FlatList
          columnWrapperStyle={
            Object {
              "flex": 1,
              "justifyContent": "space-between",
            }
          }
          data={Array []}
          disableVirtualization={false}
          extraData={
            Object {
              "arrayHolder": Array [],
              "token": "",
            }
          }
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={2}
          onEndReachedThreshold={2}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          updateCellsBatchingPeriod={50}
          windowSize={21}
        />
      </View>
    </View>
  </TouchableWithoutFeedback>
</ScrollViewMock>
`;

exports[`catalogue User navigates to catalogue 2`] = `
<ScrollViewMock
  keyboardShouldPersistTaps="always"
  style={
    Object {
      "backgroundColor": "#ffffffff",
      "flex": 1,
      "marginLeft": "auto",
      "marginRight": "auto",
      "maxWidth": 650,
      "width": "100%",
    }
  }
>
  <TouchableWithoutFeedback
    onPress={[Function]}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "flex": 1,
              "justifyContent": "center",
              "width": "100%",
            },
            Object {
              "display": "flex",
              "flexDirection": "row",
            },
          ]
        }
      >
        <FlatList
          columnWrapperStyle={
            Object {
              "flex": 1,
              "justifyContent": "space-between",
            }
          }
          data={Array []}
          disableVirtualization={false}
          extraData={
            Object {
              "arrayHolder": Array [],
              "token": "",
            }
          }
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={2}
          onEndReachedThreshold={2}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          updateCellsBatchingPeriod={50}
          windowSize={21}
        />
      </View>
    </View>
  </TouchableWithoutFeedback>
</ScrollViewMock>
`;
